### PR TITLE
[Fluent] Grid

### PR DESCRIPTION
**Pull request by Juveniel**
part of: https://github.com/telerik/kendo-themes-private/issues/3

Note: adaptive styles will be transferred with a separate PR

### CODE REVIEW

**Review by TeyaVes — CHANGES_REQUESTED** [review]

File: packages/fluent/scss/grid/index.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Module meta
+$_kendo-module-meta: (
+    name: "grid",
+    dependencies: (
+        "icon",
```
[the remark this answers is not in the document]
Comment (Juveniel) [reply packages/fluent/scss/grid/index.scss:5]: We need this as all states of the grid rows inherit from $kendo-table-state

File: packages/fluent/scss/table/_layout.scss [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        border-color: inherit;
+        text-align: inherit;
+    }
+
+    .k-table-th,
+    .k-table-td {
+        padding-inline: var( --INTERNAL--kendo-table-cell-padding-x, 0 );
+        padding-block: var( --INTERNAL--kendo-table-cell-padding-y, 0 );
+        border-block-width: var( --kendo-table-cell-horizontal-border-width, #{$kendo-table-cell-horizontal-border-width} );
+        border-inline-width: var( --kendo-table-cell-vertical-border-width, #{$kendo-table-cell-vertical-border-width} );
```
[the remark this answers is not in the document]
Comment (Juveniel) [reply packages/fluent/scss/table/_layout.scss:50]: The vertical/horizontal part of the name implies what the variable is used for e.g x/y border. Thus, I would suggest not to change this.

**Review by TeyaVes — APPROVED** [review]

File: packages/fluent/scss/table/_layout.scss [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        border-style: solid;
+        box-sizing: border-box;
+
+        .k-table {
+            border-width: 0;
+        }
+    }
+
+    .k-table-header-wrap {
+        margin-inline-end: -1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_layout.scss:85]:
```suggestion
        margin-inline-end: calc( var( --kendo-table-cell-vertical-border-width, #{$kendo-table-cell-vertical-border-width} ) * -1 );
```

File: packages/fluent/scss/grid/_layout.scss [hunk 4]
```diff
@@ -0,0 +1,1267 @@
...
+        .k-grouping-row .k-icon {
+            margin-inline-start: calc( ( var( --kendo-icon-size, 1rem ) / 2 ) - #{$kendo-table-cell-padding-x} );
+            margin-inline-end: calc( var( --kendo-icon-size, 1rem ) / 2 );
+            text-decoration: none;
+        }
+
+        .k-group-footer td {
+            border-style: solid;
+            border-block-start-width: var( --kendo-grid-group-footer-border-width-y, #{$kendo-grid-group-footer-border-width-y} );
+            border-inline-width: var( --kendo-grid-group-footer-border-width-x, #{$kendo-grid-group-footer-border-width-x} );
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:150]:
```suggestion
            border-inline-width: var( --kendo-grid-group-footer-border-width-x, #{$kendo-grid-group-footer-border-width-x} );
            border-block-start-width: var( --kendo-grid-group-footer-border-width-y, #{$kendo-grid-group-footer-border-width-y} );
```

File: packages/fluent/scss/grid/_layout.scss [hunk 5]
```diff
@@ -0,0 +1,1267 @@
...
+            top: var( --kendo-grid-group-dropclue-size, #{$kendo-grid-group-dropclue-size} );
+            left: calc( var( --kendo-grid-group-dropclue-size, #{$kendo-grid-group-dropclue-size} ) - ( var( --kendo-grid-group-dropclue-line-size, #{$kendo-grid-group-dropclue-line-size} ) / 2 ) );
+        }
+    }
+
+
+    // Grid header / footer
+    .k-grid-header-wrap,
+    .k-grid-footer-wrap {
+        margin-inline-end: -1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:360]:
```suggestion
        margin-inline-end: calc( var( --kendo-grid-cell-border-width-x, #{$kendo-grid-cell-border-width-x} ) * -1 );
```

File: packages/fluent/scss/grid/_layout.scss [hunk 6]
```diff
@@ -0,0 +1,1267 @@
...
+
+    div.k-grid-header,
+    div.k-grid-footer {
+        display: flex;
+        flex-direction: row;
+        align-items: stretch;
+    }
+
+    .k-grid-header {
+        border-bottom-width: 1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:401]:
```suggestion
        border-bottom-width: $kendo-grid-header-border-width-y;
```

Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:401]:
```suggestion
        border-bottom-width: var( --kendo-grid-header-border-width-y, $kendo-grid-header-border-width-y );
```

File: packages/fluent/scss/grid/_layout.scss [hunk 7]
```diff
@@ -0,0 +1,1267 @@
...
+        font-size: var( --kendo-grid-header-font-size, #{$kendo-grid-header-font-size} );
+
+        table {
+            margin-bottom: -1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:405]: calc( var( --kendo-grid-header-border-width-y, $kendo-grid-header-border-width-y ) * -1 )

File: packages/fluent/scss/grid/_layout.scss [hunk 8]
```diff
@@ -0,0 +1,1267 @@
...
+
+        .k-grid-filter,
+        .k-header-column-menu,
+        .k-grid-header-menu {
+            padding: $kendo-button-padding-y-md;
+            width: $kendo-button-calc-size;
+            height: $kendo-button-calc-size;
+            box-sizing: border-box;
+            outline: 0;
+            line-height: $kendo-form-line-height;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_layout.scss:457]: missing grid and CSS variables

File: packages/fluent/scss/grid/_variables.scss [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+$kendo-grid-header-padding-y: $kendo-grid-padding-y !default;
+/// Vertical border width of the grid header.
+/// @group grid
+$kendo-grid-header-border-width-y: $kendo-grid-border-width !default;
+/// Horizontal border width of the grid header.
+/// @group grid
+$kendo-grid-header-border-width-x: 0 !default;
+/// Font size of the grid header.
+/// @group grid
+$kendo-grid-header-font-size: null !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:97]: this make null fallback

File: packages/fluent/scss/grid/_variables.scss [hunk 10]
```diff
@@ -0,0 +1,460 @@
...
+$kendo-grid-sort-index-height: var( --kendo-icon-size, 1rem ) !default;
+/// Horizontal offset of the grid sort index.
+/// @group grid
+$kendo-grid-sort-index-spacing-y: calc( var( --kendo-icon-spacing, .5rem ) / 2) !default;
+/// Vertical offset of the grid sort index.
+/// @group grid
+$kendo-grid-sort-index-spacing-x: calc( $kendo-grid-sort-index-spacing-y * -1 ) !default;
+/// Spacing of the grid sort icon.
+/// @group grid
+$kendo-grid-sort-icon-spacing: calc( #{$kendo-padding-x} - 1px ) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:215]: should be 6px by design

File: packages/fluent/scss/grid/_variables.scss [hunk 11]
```diff
@@ -0,0 +1,460 @@
...
+$kendo-grid-group-dropclue-dot-size: $kendo-grid-group-dropclue-size !default;
+/// Height of the grid  group drop clue.
+/// @group grid
+$kendo-grid-group-dropclue-height: calc( $kendo-button-calc-size + $kendo-grid-group-dropclue-size ) !default;
+/// Top offset of the grid group drop clue.
+/// @group grid
+$kendo-grid-group-dropclue-top: 2px !default;
+/// Text color of the grid group drop clue.
+/// @group grid
+$kendo-grid-group-dropclue-text: get-theme-color-var( primary-130 ) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:292]:
```suggestion
$kendo-grid-group-dropclue-bg: get-theme-color-var( primary-130 ) !default;
```

File: packages/fluent/scss/grid/_variables.scss [hunk 12]
```diff
@@ -0,0 +1,460 @@
...
+/// Alt selected background color of the grid sticky cells.
+/// @group grid
+$kendo-grid-sticky-selected-alt-bg: $kendo-grid-sticky-selected-bg !default;
+/// Selected hover background color of the grid sticky cells.
+/// @group grid
+$kendo-grid-sticky-selected-hover-bg: $kendo-grid-selected-hover-bg !default;
+
+/// Horizontal padding of the grid filter menu container.
+/// @group grid
+$kendo-filter-menu-container-padding-y: map-get( $kendo-spacing, 3) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:342]:
```suggestion
$kendo-filter-menu-container-padding-y: map-get( $kendo-spacing, 3 ) !default;
```

File: packages/fluent/scss/grid/_variables.scss [hunk 13]
```diff
@@ -0,0 +1,460 @@
...
+/// Vertical padding of the grid filter menu container.
+/// @group grid
+$kendo-filter-menu-container-padding-x: map-get( $kendo-spacing, 3) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:345]:
```suggestion
$kendo-filter-menu-container-padding-x: map-get( $kendo-spacing, 3 ) !default;
```

File: packages/fluent/scss/grid/_variables.scss [hunk 14]
```diff
@@ -0,0 +1,460 @@
...
+
+/// Default width of the grid column menu.
+/// @group grid
+$kendo-grid-column-menu-width: 230px !default;
+/// Spacing of the grid column menu.
+/// @group grid
+$kendo-grid-column-menu-spacing: map-get( $kendo-spacing, 2) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:352]:
```suggestion
$kendo-grid-column-menu-spacing: map-get( $kendo-spacing, 2 ) !default;
```

File: packages/fluent/scss/grid/_variables.scss [hunk 15]
```diff
@@ -0,0 +1,460 @@
...
+/// Text color of the grid column menu group.
+/// @group grid
+$kendo-grid-column-menu-group-header-text: var( --kendo-component-text, inherit ) !default;
+/// Border color of the grid column menu group.
+/// @group grid
+$kendo-grid-column-menu-group-header-border: var( --kendo-component-border, inherit) !default;
+
+/// Default width of the grid drag cell.
+/// @group grid
+$kendo-grid-drag-cell-width: calc( var( --kendo-icon-size, 1rem ) * 2.25) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:454]:
```suggestion
$kendo-grid-drag-cell-width: calc( var( --kendo-icon-size, 1rem ) * 2.25 ) !default;
```

File: packages/fluent/scss/grid/_variables.scss [hunk 16]
```diff
@@ -0,0 +1,460 @@
...
+/// Default width of the grid hierarchy cell.
+/// @group grid
+$kendo-grid-hierarchy-col-width: calc( var( --kendo-icon-size, 1rem ) * 2) !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/grid/_variables.scss:457]:
```suggestion
$kendo-grid-hierarchy-col-width: calc( var( --kendo-icon-size, 1rem ) * 2 ) !default;
```

File: packages/fluent/scss/table/_layout.scss (hunk 3 above)
Comment (magdalenaan) [inline packages/fluent/scss/table/_layout.scss:85]:
```suggestion
        margin-inline-end: calc( var( --kendo-table-cell-vertical-border-width, #{$kendo-table-cell-vertical-border-width} ) * -1 );
```

File: packages/fluent/scss/table/_layout.scss [hunk 17]
```diff
@@ -0,0 +1,289 @@
...
+        border-inline-end-width: var( --kendo-table-cell-vertical-border-width, #{$kendo-table-cell-vertical-border-width} );
+        border-style: solid;
+        border-color: inherit;
+        overflow: hidden;
+    }
+
+    .k-table-header,
+    .k-table-header-wrap {
+        > .k-table {
+            margin-block-end: -1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_layout.scss:98]: use negative value of the border-width variable

File: packages/fluent/scss/table/_layout.scss [hunk 18]
```diff
@@ -0,0 +1,289 @@
...
+        .k-table {
+            border-width: 0;
+        }
+
+        .k-table-td {
+            border-block-width: 0;
+        }
+    }
+    .k-table-footer-wrap {
+        margin-inline-end: -1px;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_layout.scss:241]: use negative value of the border-width variable

File: packages/fluent/scss/table/_variables.scss [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+/// Text color of hovered rows in table.
+/// @group table
+$kendo-table-hover-text: var( --kendo-hover-text, inherit ) !default;
+/// Border color of hovered rows in table.
+/// @group table
+$kendo-table-hover-border: var( --kendo-hover-border, inherit ) !default;
+
+/// Background color of focused rows in table.
+/// @group table
+$kendo-table-focus-bg: null !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_variables.scss:140]: this make fallback value to be null

File: packages/fluent/scss/table/_variables.scss [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+/// Text color of focused rows in table.
+/// @group table
+$kendo-table-focus-text: null !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_variables.scss:143]: the same as above

File: packages/fluent/scss/table/_variables.scss [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+/// Border color of focused rows in table.
+/// @group table
+$kendo-table-focus-border: null !default;
```
Comment (magdalenaan) [inline packages/fluent/scss/table/_variables.scss:146]: the same as above

**Comment by Juveniel** [comment]
All changes are addressed

**Pull request merged**